python-package/lightgbm/engine.py

```diff
@@ -299,6 +328,90 @@ def handler_function(*args, **kwargs):
             return ret
         return handler_function
 
+    def __getstate__(self) -> Dict[str, Any]:
+        return vars(self)
+
+    def __setstate__(self, state: Dict[str, Any]) -> None:
+        vars(self).update(state)
+
+    def model_from_string(self, model_str: str) -> "CVBooster":
+        """Load CVBooster from a string.
+
+        Parameters
+        ----------
+        model_str : str
+            Model will be loaded from this string.
+
+        Returns
+        -------
+        self : CVBooster
+            Loaded CVBooster object.
+        """
+        self._from_dict(json.loads(model_str))
+        return self
```

Comment:
@jameslamb 
These method names are given to override `model_from_string` and `model_to_string` of `Booster`. I agree that the class method is cleaner code for these use cases, but users may be confused because both `CVBooster.from_string` and `CVBooster.model_from_string` are callable.

I think it would be better to leave the method names as they are and specify in the docstring which methods are actually overridden, what do you think?


draft of docstring:
```
Auxiliary data structure to hold and redirect all boosters of ``cv`` function.
This class has the same methods as Booster class.
All method calls, except for the following methods, are actually performed for underlying Boosters and 
then all returned results are returned in a list.

- model_from_string
- model_to_string
- save_model
```

Reply:
I agree with @nyanp and assumed that was the reason when I reviewed this...`Booster` has these methods as instance methods, so I think `CVBooster` should as well.

If we were re-designing these classes from scratch I'd agree that class methods are preferable, but I don't think such aa breaking change is worth doing now.

Reply:
And I agree with the proposed docstring @nyanp

Reply:
Oops, sorry, I was replying to another account... Thanks for the suggestion @jmoralez , and thanks for your comment @jameslamb !

> If we were re-designing these classes from scratch I'd agree that class methods are preferable, but I don't think such aa breaking change is worth doing now.

Yes, I agree. I will modify the docstring ;)